index.js

```diff
@@ -338,6 +345,34 @@ class TrezorKeyring extends EventEmitter {
     this.paths = {};
   }
 
+  /**
+   * Sets the hdPath property to the passed param, if it is an explicitly supported hdPath.
+   * If the passed param is equal to the current hdPath, then this method has no effect. If it
+   * is different, the hdPath is updated, and the hdk, accounts, page and perPage properties are
+   * reset.
+   *
+   * If the passed hdPath is not explicitly supported, an error is thrown.
```

Comment:
This description reads like I was explaining the code line-by-line. I don't think it's relevant here which specific properties are reset for example. The point is that the state of the keyring is reset - whatever that state may consist of. 

Here's an attempt to describe the purpose of the method for consumers, without referencing internals.

```suggestion
   * Set the HD path to be used by the keyring. Only known supported HD paths are allowed.
   * 
   * If the given HD path is already the current HD path, nothing happens. Otherwise the new HD
   * path is set, and the wallet state is completely reset.
   *
   * @throws {Error] Throws if the HD path is not supported.
```